src/binaryutils.cpp

```diff
@@ -322,6 +356,63 @@ Status convertTensorToMatsMatchingTensorInfo(const tensorflow::TensorProto& src,
     return StatusCode::OK;
 }
 
+Status convertInferTensorContentsToMatsMatchingTensorInfo(const ::inference::ModelInferRequest::InferInputTensor& src, std::vector<cv::Mat>& images, const std::shared_ptr<TensorInfo>& tensorInfo) {
+    Dimension targetHeight = getTensorInfoHeightDim(tensorInfo);
+    Dimension targetWidth = getTensorInfoWidthDim(tensorInfo);
+
+    // Enforce resolution alignment against first image in the batch if resize is not supported.
+    bool resizeSupported = isResizeSupported(tensorInfo);
+    bool enforceResolutionAlignment = !resizeSupported;
+
+    for (int i = 0; i < src.contents().bytes_contents_size(); i++) {
+        cv::Mat image = convertStringToMat(src.contents().bytes_contents(i));
+        if (image.data == nullptr)
```

Comment:
Could we templatize this method to avoid doubling the logic? Just create helper structure that have specialized implementations per underlying type to extract i-th buffer.